rosbag2_storage/src/impl/storage_factory_impl.hpp

```diff
@@ -54,13 +54,17 @@ std::shared_ptr<InterfaceT>
 get_interface_instance(
   std::shared_ptr<pluginlib::ClassLoader<InterfaceT>> class_loader,
   const std::string & storage_id,
-  const std::string & uri)
+  const std::string & uri,
+  bool last_attempt)
 {
   const auto & registered_classes = class_loader->getDeclaredClasses();
   auto class_exists = std::find(registered_classes.begin(), registered_classes.end(), storage_id);
   if (class_exists == registered_classes.end()) {
-    RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME,
-      "Requested storage id %s does not exist", storage_id.c_str());
+    // This prevents unnecessary error messages when loading a ReadWrite storage as ReadOnly storage
+    if (last_attempt) {
```

Comment:
what's the exact purpose of this? When is decided that this is a last attempt? Can you elaborate on this?

Reply:
Sure: Before this change, whenever you used `rosbag_play`, you'd get an error message "Requested storage id sqlite3 does not exist" because there is no ReadOnlyStorage `sqlite3` from pluginlib's perspective. Obviously, since there is a ReadWriteStorage `sqlite3`, we still open the storage and everything works correctly, but the error message is highly irritating.